### PR TITLE
Added documentation on adding javaagent to container

### PR DESCRIPTION
**Pull request by CauchyPeano**
Please see #1537

### CODE REVIEW

File: docs/faq.md [hunk 1]
```diff
@@ -415,7 +415,33 @@ jib.container.useCurrentTimestamp = true
...
+```groovy
+jib {
+    container {
+        jvmFlags = ['-javaagent:/myfolder/agent.jar']
+    }
+}
+
+```
 
 *TODO: Provide more comprehensive solution.*
```
Comment (TadCordle) [inline docs/faq.md]: Maybe we can remove this TODO now.

File: docs/faq.md [hunk 2]
```diff
@@ -415,7 +415,33 @@ jib.container.useCurrentTimestamp = true
 
 ### I would like to run my application with a javaagent.
 
-See [Can I ADD a custom directory to the image?](#can-i-add-a-custom-directory-to-the-image).
+See also [Can I ADD a custom directory to the image?](#can-i-add-a-custom-directory-to-the-image).
```
Comment (TadCordle) [inline docs/faq.md]: Might make more sense to put this line at the bottom of this section now.

File: docs/faq.md [hunk 3]
```diff
@@ -415,7 +415,33 @@ jib.container.useCurrentTimestamp = true
...
+
+You need to put your javaagent into your `src/main/jib` folder, so that this file will end in your containers file system.
```
Comment (TadCordle) [inline docs/faq.md]: "You can add a javaagent to your container's file system by placing it in the `src/main/jib` folder then pointing to it with Jib's `container.jvmFlags` configuration." might be cleaner?

File: docs/faq.md [hunk 4]
```diff
@@ -415,7 +415,33 @@ jib.container.useCurrentTimestamp = true
...
+      <jvmFlag>-javaagent:/myfolder/agent.jar</jvmFlag>
+    </jvmFlags>    
+  </container>
+</configuration>
+```
+
+#### Gradle
+
+```groovy
+jib {
```
Comment (chanseokoh) [inline docs/faq.md]: I'd go with `jib.container.jvmFlags` for brevity.

**Comment by briandealwis** [comment]
There is an example in [`examples/java-agent`](https://github.com/GoogleContainerTools/jib/tree/master/examples/java-agent) but it installs the Stackdriver Debugging agent which has some more complex setup requirements.  We can reference it at the end of this new section.

File: docs/faq.md (hunk 3 above)
Comment (loosebazooka) [reply docs/faq.md]: This sentence seems to suggest that you drop it directly in `src/main/jib`, perhaps we should say "placing it *somewhere* in the `src/main/jib` directory" or something. Also **folder** is more of a UI construct, would prefer we use **directory** here.

File: docs/faq.md [hunk 5]
```diff
@@ -415,9 +415,27 @@ jib.container.useCurrentTimestamp = true
 
 ### I would like to run my application with a javaagent.
 
-See [Can I ADD a custom directory to the image?](#can-i-add-a-custom-directory-to-the-image).
+You can run your container with a javaagent by placing it in the `src/main/jib` folder to add it to the container's filesystem, then pointing to it using Jib's `container.jvmFlags` configuration.
```
Comment (TadCordle) [inline docs/faq.md]: >This sentence seems to suggest that you drop it directly in `src/main/jib`, perhaps we should say "placing it **somewhere** in `the src/main/jib` directory" or something. Also **folder** is more of a UI construct, would prefer we use **directory** here.

Comment (CauchyPeano) [reply docs/faq.md]: yes, true, will update this one.

**Comment by loosebazooka** [comment]
I triggered a new macos build, I'll merge then when it's done.

**Comment by chanseokoh** [comment]
Thanks for your contribution!

**Pull request merged**